spec/sailor.spec.js

```diff
@@ -224,6 +224,55 @@ describe('Sailor', function () {
                 .catch(done); //todo: use done.fail after migration to Jasmine 2.x
         });
 
+        it('should call sendData() and ack() only once', function (done) {
+            settings.FUNCTION = 'end_after_data_twice';
+            const sailor = new Sailor(settings);
+
+            spyOn(sailor.apiClient.tasks, 'retrieveStep').andCallFake(function(taskId, stepId) {
+                expect(taskId).toEqual('5559edd38968ec0736000003');
+                expect(stepId).toEqual('step_1');
+                return Q({});
+            });
+
+            sailor.connect()
+                .then(() => sailor.prepare())
+                .then(() => sailor.processMessage(payload, message))
+                .then(() => {
+                    expect(sailor.apiClient.tasks.retrieveStep).toHaveBeenCalled();
+                    expect(fakeAMQPConnection.connect).toHaveBeenCalled();
+                    expect(fakeAMQPConnection.sendData).toHaveBeenCalled();
+
+                    const sendDataCalls = fakeAMQPConnection.sendData.calls;
+
+                    expect(sendDataCalls[0].args[0]).toEqual({items: [1,2,3,4,5,6]});
+                    expect(sendDataCalls[0].args[1]).toEqual(jasmine.any(Object));
+                    expect(sendDataCalls[0].args[1]).toEqual({
```

Comment:
This test case proclaims about number of calls of `ack` and `sendData` so assertion for args for `sendData` seems redundant and makes all test suite more fragile.